luaui/Widgets/gui_selfd_icons.lua

```diff
@@ -209,8 +209,13 @@ function widget:UnitCommand(unitID, unitDefID, teamID, cmdID, cmdParams, cmdOpts
 	end
 
 	if cmdID ~= CMD.SELFD and not cmdOpts.shift and queuedSelfD[unitID] then
-		-- had a queued selfd, but the queue was replaced, so mark not queued
-		queuedSelfD[unitID] = nil
+		-- had a queued selfd, but the queue was potentially replaced
+
+		-- check for commands that don't replace the queue
+		if cmdID ~= CMD.REMOVE and cmdID ~= CMD.INSERT then
```

Comment:
feedback especially welcome on this list of commands

Reply:
anything stateful (cmd.firestate, cmd.onoff etc), cmd.wait (which isn't stateful!)

Reply:
fixed, thank you!